merlin/dag/ops/subgraph.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+from typing import Callable
+
+from merlin.core.protocols import Transformable
+from merlin.dag.executors import DaskExecutor, LocalExecutor
+from merlin.dag.graph import Graph
+from merlin.dag.selector import ColumnSelector
+from merlin.dag.stat_operator import StatOperator
+from merlin.io.dataset import Dataset
+from merlin.schema import Schema
+
+
+class Subgraph(StatOperator):
+    """
+    Operator that executes a Merlin Graph as a sub-graph of another Graph
+    """
+
+    def __init__(self, name, output_node, loop_until: Callable = None):
+        self.name = name
+        self.graph = output_node
+        self.loop_until = loop_until
+
+        if not isinstance(output_node, Graph):
+            self.graph = Graph(output_node)
+
+        super().__init__()
+
+    def transform(
+        self, col_selector: ColumnSelector, transformable: Transformable
+    ) -> Transformable:
+        """Simply returns the selected output columns from the input dataframe
+
+        The main functionality of this operator has to do with computing the schemas
+        for selection nodes in the Workflow graph, so very little has to happen in the
+        `transform` method.
+
+        Parameters
+        -----------
+        columns: list of str or list of list of str
+            The columns to apply this operator to
+        transformable: Transformable
+            A pandas or cudf dataframe that this operator will work on
+
+        Returns
+        -------
+        DataFrame
+            Returns a transformed dataframe for this operator
+        """
+        executor = LocalExecutor()
+        transformable = executor.transform(transformable, self.graph)
+        while self.loop_until and not self.loop_until(transformable):
+            transformable = executor.transform(transformable, self.graph)
+        return transformable
+
+    def fit(self, col_selector: ColumnSelector, dataset: Dataset):
+        DaskExecutor().fit(dataset, self.graph)
```

Comment:
It feels a bit unexpected that the `transform` uses `LocalExecutor`, while the `fit` uses `DaskExecutor`. Maybe don't need to action this in this PR, but something doesn't feel right about that. Are we thinking of extending support for `fit` beyond the dask versions? The addition of the StatOperator in this PR suggests we might be?